tests/baselines/reference/tsserver/codeFix/install-package.js

```diff
@@ -0,0 +1,462 @@
+Info seq  [hh:mm:ss:mss] 	Files (2)
+
+Info seq  [hh:mm:ss:mss] -----------------------------------------------
+Info seq  [hh:mm:ss:mss] Open files: 
+Info seq  [hh:mm:ss:mss] 	FileName: /home/src/projects/project/src/file.ts ProjectRootPath: undefined
+Info seq  [hh:mm:ss:mss] 		Projects: /home/src/projects/project/tsconfig.json
+Info seq  [hh:mm:ss:mss] response:
+    {
+      "responseRequired": false
+    }
+After request
+
+PolledWatches::
+/home/src/projects/node_modules: *new*
+  {"pollingInterval":500}
+/home/src/projects/node_modules/@types: *new*
+  {"pollingInterval":500}
+/home/src/projects/project/node_modules/@types: *new*
+  {"pollingInterval":500}
+
+FsWatches::
+/a/lib/lib.d.ts: *new*
+  {}
+/home/src/projects/project/tsconfig.json: *new*
+  {}
+
+FsWatchesRecursive::
+/home/src/projects/project: *new*
+  {}
+/home/src/projects/project/node_modules: *new*
+  {}
+/home/src/projects/project/src: *new*
+  {}
+
+Before request
+
+Info seq  [hh:mm:ss:mss] request:
+    {
+      "command": "getCombinedCodeFix",
+      "arguments": {
+        "scope": {
+          "type": "file",
+          "args": {
+            "file": "/home/src/projects/project/src/file.ts"
+          }
+        },
+        "fixId": "installTypesPackage"
+      },
+      "seq": 2,
+      "type": "request"
+    }
+TI:: Creating typing installer
+
+TI:: [hh:mm:ss:mss] Global cache location '/a/data', safe file path '/safeList.json', types map path /typesMap.json
+TI:: [hh:mm:ss:mss] Processing cache location '/a/data'
+TI:: [hh:mm:ss:mss] Trying to find '/a/data/package.json'...
+TI:: [hh:mm:ss:mss] Finished processing cache location '/a/data'
+TI:: [hh:mm:ss:mss] Npm config file: /a/data/package.json
+TI:: [hh:mm:ss:mss] Npm config file: '/a/data/package.json' is missing, creating new one...
+TI:: [hh:mm:ss:mss] Updating types-registry npm package...
+TI:: [hh:mm:ss:mss] npm install --ignore-scripts types-registry@latest
+TI:: [hh:mm:ss:mss] Updated types-registry npm package
+TI:: typing installer creation complete
+//// [/a/data/package.json]
+{ "private": true }
+
+//// [/a/data/node_modules/types-registry/index.json]
+{
+  "entries": {
+    "vscode": {
+      "latest": "1.3.0",
+      "ts2.0": "1.0.0",
+      "ts2.1": "1.0.0",
+      "ts2.2": "1.2.0",
+      "ts2.3": "1.3.0",
+      "ts2.4": "1.3.0",
+      "ts2.5": "1.3.0",
+      "ts2.6": "1.3.0",
+      "ts2.7": "1.3.0"
+    }
+  }
+}
+
+
+TI:: [hh:mm:ss:mss] Sending response:
+    {
+      "kind": "event::typesRegistry",
+      "typesRegistry": {
+        "vscode": {
+          "latest": "1.3.0",
+          "ts2.0": "1.0.0",
+          "ts2.1": "1.0.0",
+          "ts2.2": "1.2.0",
+          "ts2.3": "1.3.0",
+          "ts2.4": "1.3.0",
+          "ts2.5": "1.3.0",
+          "ts2.6": "1.3.0",
+          "ts2.7": "1.3.0"
+        }
+      }
+    }
+Info seq  [hh:mm:ss:mss] response:
+    {
+      "response": {
+        "changes": [],
+        "commands": [
+          {
```

Comment:
Should these be deduplicated? If yes, at the end or during addition?
 (irrespective of that we do need the fix to TI adapter to handle the correct promise resolution)

Reply:
Probably? I'm not familiar enough with this to know for sure...

Reply:
May be @sandersn

Reply:
I'm guessing that this will run two `npm installs`, at once probably, and should be deduped. It *seems* safer, and just as easy, to do it on addition, but I could be wrong. It looks like the key of the new Map might need to be based on the package name instead of a numeric id.

@zkat worked on this code much more recently than I have and might have additional thoughts.

Reply:
its possible to have duplicate "npm install requests" if i clicked "fix all" command multiple times right? So  it has to be numeric id and installer handles the multiple installs for same package

Reply:
The deduped result would be nice but wondering if it really matters

Reply:
It'd be equivalent to running two `npm install X` commands at once, right? If a manual test of that shows that npm handles that case on its own, I think it's fine to have ATA allow dupes.

Reply:
Its does